src/tempo_config.py

```diff
@@ -24,143 +22,184 @@ class InvalidConfigurationError(Exception):
     pass
 
 
-class Memberlist(BaseModel):
-    """Memberlist schema."""
-
-    cluster_label: str
-    cluster_label_verification_disabled: bool = False
-    join_members: List[str]
-
-
-class Tsdb(BaseModel):
-    """Tsdb schema."""
+class Kvstore(BaseModel):
+    """Kvstore schema."""
 
-    dir: str = "/data/ingester"
+    store: str = "memberlist"
 
 
-class BlocksStorage(BaseModel):
-    """Blocks storage schema."""
+class Ring(BaseModel):
+    """Ring schema."""
 
-    storage_prefix: str = "blocks"
-    tsdb: Tsdb
+    kvstore: Kvstore
+    replication_factor: int
 
 
-class Limits(BaseModel):
-    """Limits schema."""
+class Memberlist(BaseModel):
+    """Memberlist schema."""
 
-    ingestion_rate: int = 0
-    ingestion_burst_size: int = 0
-    max_global_series_per_user: int = 0
-    ruler_max_rules_per_rule_group: int = 0
-    ruler_max_rule_groups_per_tenant: int = 0
+    abort_if_cluster_join_fails: bool
+    bind_port: int
+    join_members: List[str]
+    tls_enabled: Optional[bool] = False
```

Comment:
is this truly optional? as in, can it be None? or is it really bool (given that the default is False...)

Reply:
its semantically misleading. Will make it a bool